### PR TITLE
Add OpenFunction client

### PR DESCRIPTION
**Pull request by wentevill**
fix issue #174

### CODE REVIEW

**Comment by wentevill** [comment]
some CI is wrong, but seems not related to this PR.
```
 cat config/samples/function-sample-serving-only.yaml | kubectl apply -f -
  shell: /usr/bin/bash -e {0}
  env:
    OF_IMAGE: openfunction/openfunction:latest
    REGISTRY_SERVER: https://index.docker.io/v1/
    REGISTRY_USER: 
    REGISTRY_PASSWORD: 
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "mfunctions.of.io": Post "https://openfunction-webhook-service.openfunction.svc:443/mutate-core-openfunction-io-v1alpha2-function?timeout=10s": dial tcp 10.96.223.123:443: connect: connection refused
```

**Comment by benjaminhuo** [comment]
Thanks for the PR, and yes @wanjunlei will fix the CI error and it's not relevant to this PR

File: apis/core/v1alpha1/groupversion_info.go [hunk 1]
```diff
@@ -25,11 +25,11 @@ import (
 )
 
 var (
-	// GroupVersion is group version used to register these objects
-	GroupVersion = schema.GroupVersion{Group: "core.openfunction.io", Version: "v1alpha1"}
+	// SchemeGroupVersion is group version used to register these objects
```
Comment (benjaminhuo) [inline apis/core/v1alpha1/groupversion_info.go:28]: I think we can keep GroupVersion as it is and add SchemeGroupVersion for client as keda does https://github.com/kedacore/keda/blob/main/apis/keda/v1alpha1/groupversion_info.go

Comment (wentevill) [reply apis/core/v1alpha1/groupversion_info.go:28]: This is a very good suggestion, i will modify it like this.

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
Then we can import these client pkg in the cli project to replace existing method in cli @wentevill

**Pull request merged**